src/widget/form/settings/advancedform.cpp

```diff
@@ -30,16 +40,32 @@ AdvancedForm::AdvancedForm() :
     bodyUI = new Ui::AdvancedSettings;
     bodyUI->setupUi(this);
 
+    Settings &s = Settings::getInstance();
+    bodyUI->cbEnableIPv6->setChecked(s.getEnableIPv6());
     bodyUI->cbMakeToxPortable->setChecked(Settings::getInstance().getMakeToxPortable());
+    bodyUI->cbEnableUDP->setChecked(!s.getForceTCP());
+    bodyUI->proxyAddr->setText(s.getProxyAddr());
+    int port = s.getProxyPort();
+    if (port != -1)
+        bodyUI->proxyPort->setValue(port);
```

Comment:
Port should become a `quint16` (`int port = s.getProxyPort()`), where 0 already is the invalid port. Further the value must not be validated here! Settings values are just stored/loaded. Validation takes place when using those values.